external/esp_idf_port/esp32/esp32_wifi_utils.c

```diff
@@ -245,13 +246,20 @@ static esp_err_t event_handler(void *ctx, system_event_t *event)
 		s_retry_num++;
 		int reason = event->event_info.disconnected.reason;
 		ESP_LOGI(TAG, "[%s] connect to the AP %s fail %d: %d; %d\n", __func__, event->event_info.disconnected.ssid, s_retry_num, event->event_info.disconnected.reason, isStaConnected);
-    
+        
         /*give chance to other thread to continue*/
         sleep(1);
 		if (s_retry_num > 0) {
 			ESP_LOGI(TAG, "g_cbk.sta_connected fail\n");
-			g_cbk.sta_connected(WIFI_UTILS_FAIL, NULL);
-			s_retry_num = 0;
+            if (s_retry_num < WIFI_CONN_RETRY) {
```

Comment:
please check if below scenario is possible：
after 'wifi_utils_disconnect_ap' invoked,  'SYSTEM_EVENT_STA_DISCONNECTED' fired multiple times from driver due to somewhat reason.

 if so, then ap is retried wrongly.

Reply:
tks, reset s_retry_num to fix this case